src/PowerShellEditorServices/Session/PromptNest.cs

```diff
@@ -0,0 +1,559 @@
+using System.Collections.Concurrent;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.PowerShell.EditorServices.Utility;
+
+namespace Microsoft.PowerShell.EditorServices.Session
+{
+    using System;
+    using System.Management.Automation;
+
+    /// <summary>
+    /// Represents the stack of contexts in which PowerShell commands can be invoked.
+    /// </summary>
+    internal class PromptNest : IDisposable
+    {
+        private ConcurrentStack<PromptNestFrame> _frameStack;
```

Comment:
The style here is inconsistent with other style. Personally I prefer underscores for private variables but other code that was changed did not use underscores...

Reply:
Yeah, we don't really have any defined style rules or anything. But for new files, I use the PowerShell repo's style guide, and for existing files I try to maintain the existing style of the file.

Reply:
I also prefer the underscores. It's hard to justify sweeping through the repo and changing it all at once, but my current policy is that if I touch a file and add/change a private variable, I change them all over. I'm personally in favour of new private variables using an underscore and tolerating the inconsistency as an interim phenomenon.